src/commands/functions/create.js

```diff
@@ -301,12 +303,50 @@ const downloadFromURL = async function (context, flags, args, functionsDir) {
   }
 }
 
-const installDeps = function (functionPath) {
-  return new Promise((resolve) => {
-    cp.exec('npm i', { cwd: path.join(functionPath) }, () => {
-      resolve()
-    })
-  })
+// Takes an existing list of packages and a list of packages needed by a
+// function, and computes the list of npm modules with version range that
+// must be installed.
+const getNpmInstallPackages = (existingPackages = {}, neededPackages = {}) =>
+  Object.entries(neededPackages)
+    .filter(([name]) => existingPackages[name] === undefined)
+    .map(([name, version]) => `${name}@${version}`)
+
+// When installing a function's dependencies, we first try to find a site-level
+// `package.json` file. If we do, we look for any dependencies of the function
+// that aren't already listed as dependencies of the site and install them. If
+// we don't do this check, we may be upgrading the version of a module used in
+// another part of the project, which we don't want to do.
+const installDeps = async ({ functionPackageJson, functionPath, functionsDir }) => {
+  // eslint-disable-next-line import/no-dynamic-require, node/global-require
+  const { dependencies: functionDependencies, devDependencies: functionDevDependencies } = require(functionPackageJson)
+  const sitePackageJson = await findUp('package.json', { cwd: functionsDir, type: 'file' })
+
+  // If there is no site-level `package.json`, we fall back to the old behavior
+  // of keeping that file in the function directory and running `npm install`
+  // from there.
+  if (!sitePackageJson) {
+    await execa('npm', ['i'], { cwd: functionPath })
+
+    return
+  }
+
+  // eslint-disable-next-line import/no-dynamic-require, node/global-require
+  const { dependencies: siteDependencies, devDependencies: siteDevDependencies } = require(sitePackageJson)
+  const dependencies = getNpmInstallPackages(siteDependencies, functionDependencies)
+  const devDependencies = getNpmInstallPackages(siteDevDependencies, functionDevDependencies)
+  const npmInstallPath = path.dirname(sitePackageJson)
+
+  if (dependencies.length !== 0) {
+    await execa('npm', ['i', dependencies.join(' '), '--save'], { cwd: npmInstallPath })
```

Comment:
```suggestion
    await execa('npm', ['i', ...dependencies, '--save'], { cwd: npmInstallPath })
```

Reply:
Done in https://github.com/netlify/cli/pull/3125/commits/fd8aafd709fc6751bdb5e62df454c534afdba014.

Reply:
Note: I forgot to comment, but this was not a stylistic issue, but an escaping one. I believe the previous code would not have worked with multiple dependencies, because `npm install` would have received a single argument with spaces in it, as opposed to multiple arguments.